lib/http/features/instrumentation.rb

```diff
@@ -21,7 +27,9 @@ def initialize(instrumenter: NullInstrumenter.new, namespace: "http")
       end
 
       def wrap_request(request)
-        instrumenter.instrument("start_#{name}", :request => request)
+        # Emit a separate "start" event, so a logger can print the request
+        # being run without waiting for a response
+        instrumenter.instrument("start_#{name}", :request => request) {}
```

Comment:
I dislike this part. I see reason behind this but it feels like trying to use instrumenter for something it's not supposed to be used.

Reply:
It's a common pattern in Rails itself: https://github.com/rails/rails/blob/b2eb1d1c55a59fee1e6c4cba7030d8ceb524267c/actionpack/lib/action_controller/metal/instrumentation.rb#L30-L32

I'm not a big fan of it either (I wish the `instrument` call would just automatically emit events at both the start and end of the block), but I assume this is how it was intended to be used.

Reply:
Doesn't it? https://github.com/rails/rails/blob/405aef3dede8938c8eb9789f6d6722864faddb09/activesupport/lib/active_support/notifications/instrumenter.rb#L21-L29
line 21 - start
line 29 - finish

Reply:
@joshuaflanagan I dug through the overly-complicated code for a few minutes, and I can't figure out why, but the subscribers don't receive any calls at the start of the block, just the end.

Reply:
Ok, you're right. I misinterpreted that code. Line 21 is marking the start of the event, but nothing is published until the finish on line 29.

Confirmed in the docs (https://guides.rubyonrails.org/active_support_instrumentation.html#creating-custom-events):
> Simply call instrument with a name, payload and a block. The notification will be sent after the block returns.